docs/developer/Menu-Commands.md

```diff
@@ -0,0 +1,72 @@
+# Adding Commands to Zowe Explorer Menus
+
+This page contains developer guidance specific to adding commands in core Zowe Explorer. For more general guidance on how to add a command in VS Code, see the [VS Code Commands Extension Guide](https://code.visualstudio.com/api/extension-guides/command).
+
+## Contents
+
+- [Defining a new command](#defining-a-new-commmand)
+- [Contributing menu items](#contributing-menu-items)
+- [Menu group naming conventions](#menu-group-naming-conventions)
+
+## Defining a new commmand
+
+In order to add a command to a core Zowe Explorer menu, the command must first be defined in `packages/zowe-explorer/package.json`, under [`contributes.commands`](https://code.visualstudio.com/api/references/contribution-points#contributes.commands). The following conventions should be used when defining the respective command properties:
+
+- **`command`** property: For each Zowe Explorer view in which a command can be used (i.e. Data Sets, USS, or Jobs view), one of the prefixes in the table below should be added to the front of the `command` property to indicate what Zowe Explorer view the command applies to.
+
+  | Prefix                | Applicable view | Example usage           |
+  | --------------------- | --------------- | ----------------------- |
+  | `zowe.<command>`      | Data Sets       | `zowe.addFavorite`      |
```

Comment:
or zowe.mvs to be consistent with the group naming

Reply:
@jellypuno I agree that Data Sets should have some view-specific prefix, but I was worried I might accidentally break some code since this would involve changing quite a few command names. I actually also prefer `zowe.ds` over `mvs`, since it more intuitively matches the "Data Sets" view name.

Reply:
Do you think it would be worth opening a different issue on using zowe.ds so we can test it? At least research on how big the change would be? 

Worst case scenario, We can implement the change next PI and just update this doc. What do you think?

Reply:
I think the best way to approach this is in two steps:
1. Merge in PR #1340, which is focused on taking care of command updates that impact only `package.json`.
2. Once #1340 is in, we can make another PR to add the `zowe.ds` prefix to the data set command names, make any necessary related updates in the source code, and update this menu commands documentation accordingly.

@jellypuno How does this plan sound to you?

Reply:
@lauren-li Sounds like a good plan. So update #1332 first then I'll merge it. After that merge #1340 (after it is reviewed)

Reply:
I have updated this PR to use `zowe.ds` instead of `zowe.mvs` in commit 827f1ec.